src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
@Andrews54757
> MultiOgar would become unreliable if this change took place. Think of memory leaks and seemingly random errors.
- Well try the Updates branch out to see for yourself.
> And I note that you didnt say this wasnt faster than the original multiogar.
- [Because it isn't](https://github.com/Luka967/MultiOgar-Edited/blob/master/README.md#screenshot). Maybe it's because of my changes. Maybe not. I haven't tested.
> And a single player would probably use 30 kilobytes of data in buffers. (There are more than one binarywriters used)
- > V8 is not stupid enough to keep 98KB of memory allocated and cause a memory leak.
> Using child processes effectively in nodejs is actually harder than you think
- You're underestimating me.
> BTW, some people, go crazy with minions. So dont say people wont use 4000 minions
- > If you really want to have 4000 minions in a 1920x1080 map then have fun with the errors and CPU time.
- Those players would be too stupid to change the map size.
- And nobody reported this problem for now.